### ISSUE TITLE
Custom error message not working for strings

### ISSUE THREAD

**Issue by mgreca** [ritchieanesco/json-schema-yup-transform#32]
When you define a `string` property with `format: email` or `format: uri` the validations work, but the custom error messages aren't.

```
const schema = {
  properties: {
    name: {
      type: "string",
      format: "email"
    }
  }
};

const config = {
  errors: {
    name: {
      string: "Custom Email Error!",
      format: "Testing other way",
      email: "Testing other way 2",
    }
  }
};
```

When you run the validation with that, the result will always be: `Name is an invalid email format ` and you won't be able to change the message.

Here is a working example in [code sandbox ](https://codesandbox.io/s/eager-mendel-3yr1z?file=/src/index.js:57-547)
You can see there that the required message is working, but not the other ones.

**Comment by jdpnielsen** [comment]
I believe you need to do the following:

```ts
const schema = {
  properties: {
    name: {
      type: "string",
      format: "email"
    }
  }
};

const config = {
  errors: {
    name: {
      string: "Custom Email Error!",
      format: {
        email: "Testing other way 2"
      }
    }
  }
};
```
See https://github.com/ritchieanesco/json-schema-yup-transform/blob/master/test/yup/configErrors.string.test.ts#L343-L364

@ritchieanesco I mentioned this issue here: https://github.com/ritchieanesco/json-schema-yup-transform/pull/34#discussion_r658235865

**Comment by mgreca** [comment]
@jdpnielsen thanks, it works! but I still consider it an issue, so I'll keep it open until we have a response.

**Comment by ritchieanesco** [comment]
Thanks for raising this issue.  Error messages for formatting have been changed so the format type property is no longer required in the configuration. You can find examples of these in the unit tests:

https://github.com/ritchieanesco/json-schema-yup-transform/blob/918cb9f9a64377eb63717ec5b08dc13536c0962c/test/yup/configErrors.string.test.ts#L246-L274

This change is currently available in 1.6.6-beta.1.

I'll be looking shortly to implement @jdpnielsen idea to make the messaging more flexible.
